junit5-jupiter-starter-bazel/bazeljunit5/src/main/java/com/flexport/bazeljunit5/BazelJUnit5ConsoleLauncher.java

```diff
@@ -0,0 +1,339 @@
+      throws TransformerException {
+    Transformer transformer = TransformerFactory.newInstance().newTransformer();
+    DOMSource source = new DOMSource(xmlDocument);
+    StreamResult streamResult = new StreamResult(new File(fileName));
+    transformer.transform(source, streamResult);
+  }
+
+  /**
+   * Transform args into JUnit5 fashion.
+   */
+  public static String[] transformArgs(String[] args) {
+    return transformArgsForXmlOutputFile(
+        transformArgsForTestBridgeTestOnly(
+            Arrays.asList(args), System.getenv("TESTBRIDGE_TEST_ONLY")),
+        System.getenv("XML_OUTPUT_FILE"))
+        .stream()
+        .toArray(String[]::new);
+  }
+
+  private static List<String> transformArgsForTestBridgeTestOnly(
+      List<String> args, String testBridgeTestOnly) {
+
+    if (testBridgeTestOnly == null || testBridgeTestOnly.isEmpty()) {
+      return args;
+    }
+
+    // When e.g. `bazel test --test_filter=foo //bar:test` is run, the `--test_filter=` value is
+    // set in the TESTBRIDGE_TEST_ONLY environment variable by Bazel. For that example, value of the
+    // environment variable would simply be `foo`. Here are some examples of what it might look
+    // like in practice: test.package.TestClass#testMethod, test.package.TestClass, test.package
+
+    List<String> newArgs = new ArrayList<>();
+    newArgs.addAll(filterOptions(args, Arrays.asList(SELECT_PACKAGE)));
+    newArgs.addAll(parseOptions(testBridgeTestOnly));
+
+    return newArgs;
+  }
+
+  private static List<String> transformArgsForXmlOutputFile(
+      List<String> args, String xmlOutputFile) {
+
+    if (xmlOutputFile == null || xmlOutputFile.isEmpty()) {
+      return args;
+    }
+
+    // The XML output reporting file is set in the XML_OUTPUT_FILE environment variable by JUnit4.
+    // And JUnit5 will get the reports directory from `--reports-dir`.
+
+    List<String> newArgs = new ArrayList<>(args);
+    newArgs.add("--reports-dir=" + Paths.get(xmlOutputFile).getParent().toString());
+
+    return newArgs;
+  }
+
+  /**
+   * Parse JUnit5 option from env.TESTBRIDGE_TEST_ONLY.
+   *
+   * <p>test.package: --select-package=test.package
+   *
+   * <p>test.package.TestClass: --select-class=test.package.TestClass
+   *
+   * <p>test.package.TestClass#testMethod: --select-method=test.package.TestClass#testMethod
+   *
+   * @param testBridgeTestOnly env.TESTBRIDGE_TEST_ONLY
+   * @return option
+   */
+  private static List<String> parseOptions(String testBridgeTestOnly) {
+    // transform env.TESTBRIDGE_TEST_ONLY
+
+    // test.package.TestClass#
+    if (testBridgeTestOnly.endsWith("#") || testBridgeTestOnly.endsWith("$")) {
+      testBridgeTestOnly = testBridgeTestOnly.substring(0, testBridgeTestOnly.length() - 1);
+    }
+
+    if (testBridgeTestOnly.contains("#")) {
+      // There could be multiple classes in TESTBRIDGE_TEST_ONLY which are separated by $|
+      String[] splits = testBridgeTestOnly.split("\\$\\|");
```

Comment:
`#|` is also a separator for classes without explicit methods, s.a. in: `my.pkg.TestClass#|my.pkg.TestClass2#`